luaui/Widgets/cmd_bar_hotkeys.lua

```diff
@@ -25,86 +25,55 @@ local function reloadWidgetsBindings()
--- TODO: After a period of ~6 months it should be deleted (around January 2024 or later)
-local function replaceLegacyPreset()
-	local keyFile = Spring.GetConfigString("KeybindingFile")
-	-- sometimes this file can be in other directories, we do a string match
-	local isCustom = keyFile and keyFile:match("bar_hotkeys_custom.lua") ~= nil or false
-	if not keyFile or not isCustom then
-		return false
+local function replaceDefaultWithLegacy(file)
+	if file == 'luaui/configs/hotkeys/default_keys.txt' then
+		return 'luaui/configs/hotkeys/legacy_keys.txt'
 	end
-
-
-	local newFormat = legacyToTxt[keyFile]
-	if isCustom then -- in case the dir is not root, just pass the filename not the whole path
-		newFormat = legacyToTxt["bar_hotkeys_custom.lua"]
-	end
-	if not newFormat then return false end
-
-	-- in case the user has a uikeys file already, we need to back it up because it could potentially get overwritten by the following steps
-	if VFS.FileExists("uikeys.txt") then
-		Spring.Echo("BAR Hotkeys: Found existing unused uikeys file, creating a backup called uikeys_auto_backup.txt")
-		Spring.SendCommands("keyreload")
-		os.rename("uikeys.txt", "uikeys_auto_backup.txt")
-	end
-
-	-- output the current custom .lua bindings into a uikeys.txt file
-	if isCustom then
-		Spring.Echo("BAR Hotkeys: bar_hotkeys_custom.lua found. This format is deprecated, a " .. newFormat .. " file was written to your bar folder")
-		if VFS.FileExists(keyFile) then
-			Spring.SendCommands("unbindall")
-			loadBindingsLegacy(keyFile)
-		end
-
-		Spring.SendCommands("keysave " .. newFormat)
-	else
-		Spring.SendCommands("keyreload " .. newFormat)
+	if file == 'luaui/configs/hotkeys/default_keys_60pct.txt' then
+		return 'luaui/configs/hotkeys/legacy_keys_60pct.txt'
 	end
+end
 
-	Spring.SetConfigString("KeybindingFile", newFormat)
-
-	return true
+-- if keybinds are missing, load default hotkeys
+local function fallbackToDefault(currentKeys)
+	Spring.Echo("BAR Hotkeys: Did not find keybindings file " .. currentKeys ..". Loading grid keys")
+	Spring.SendCommands("keyreload " .. keyConfig.keybindingLayoutFiles[1])
 end
 
 local function reloadBindings()
 	-- Second parameter here is just a fallback if this config is undefined
 	currentLayout = Spring.GetConfigString("KeyboardLayout", 'qwerty')
 
-	local hasLegacy = replaceLegacyPreset()
-
-	currentKeybindingsFile = Spring.GetConfigString("KeybindingFile", keyConfig.keybindingPresets["Default"])
+	currentKeybindingsFile = Spring.GetConfigString("KeybindingFile", keyConfig.keybindingLayoutFiles[1])
+	local usingOldPreset = hasOldDefaultKeys(currentKeybindingsFile)
+	if usingOldPreset then
+		currentKeybindingsFile = replaceDefaultWithLegacy(currentKeybindingsFile)
+		Spring.Echo("BAR Hotkeys: Found old default key config, replacing with legacy", currentKeybindingsFile)
+	end
 
-	if not hasLegacy then
+	-- resolve upgrading from old "default" to "legacy"
+	if usingOldPreset then
```

Comment:
If you want you can merge with before if and do an early return, i.e.:

Instead of:

```lua
if hue
  doSomething
end

if hue
  doHueSomething
else
  doElseSomething
  ```
  
  Do:
  
  ```lua
if hue
  doSomething
  doHueSomething
  return
end

  doElseSomething
  ```